src/rail_gui.cpp

```diff
@@ -1908,38 +1908,41 @@ static void SetDefaultRailGui()
 	if (_local_company == COMPANY_SPECTATOR || !Company::IsValidID(_local_company)) return;
 
 	extern RailType _last_built_railtype;
-	RailType rt = (RailType)(_settings_client.gui.default_rail_type + RAILTYPE_END);
-	if (rt == DEF_RAILTYPE_MOST_USED) {
-		/* Find the most used rail type */
-		uint count[RAILTYPE_END];
-		memset(count, 0, sizeof(count));
-		for (TileIndex t = 0; t < MapSize(); t++) {
-			if (IsTileType(t, MP_RAILWAY) || IsLevelCrossingTile(t) || HasStationTileRail(t) ||
-					(IsTileType(t, MP_TUNNELBRIDGE) && GetTunnelBridgeTransportType(t) == TRANSPORT_RAIL)) {
-				count[GetRailType(t)]++;
+	RailType rt;
+	switch (_settings_client.gui.default_rail_type) {
+		case 2: {
```

Comment:
Perhaps a separate enum would be better, rather than the magic numbers

Reply:
I've the feeling that these fixes should go into master before 1.10.0 is released :D